common/display/displayqueue.cpp

```diff
@@ -219,6 +219,7 @@ bool DisplayQueue::QueueUpdate(std::vector<HwcLayer*>& source_layers,
     overlay_layer.SetBlending(layer->GetBlending());
     overlay_layer.SetSourceCrop(layer->GetSourceCrop());
     overlay_layer.SetDisplayFrame(layer->GetDisplayFrame());
+    overlay_layer.SetSurfaceDamage(layer->GetSurfaceDamage());
```

Comment:
Thanks for claryfying things. The only thing I think we should resolve here is instead of having ValidateSurfaceDamage as a separate function, lets have it here. This would have the advantage that for now we can have the minimum case where if no layer has damage than we skip commit (like u do now) but in next steps u can easily modify it to swap previous and current layer buffer here itself and not having to worry about modifying anything in the followup steps